pkg/apis/v1/nodepool_conversion.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"strings"
+
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	"sigs.k8s.io/karpenter/pkg/operator/injection"
+)
+
+// Convert v1 NodePool to v1beta1 NodePool
+func (in *NodePool) ConvertTo(ctx context.Context, to apis.Convertible) error {
+	v1beta1NP := to.(*v1beta1.NodePool)
+	v1beta1NP.Name = in.Name
+	v1beta1NP.UID = in.UID
+	v1beta1NP.Labels = in.Labels
+	v1beta1NP.Annotations = in.Annotations
+	in.Spec.convertTo(ctx, &v1beta1NP.Spec)
+
+	// Convert v1 status
+	v1beta1NP.Status.Resources = in.Status.Resources
+
+	v1beta1NP.Annotations = lo.Assign(v1beta1NP.Annotations, map[string]string{
+		NodePoolHashVersion:              v1beta1NP.Hash(),
+		NodePoolHashVersionAnnotationKey: v1beta1.NodePoolHashVersion,
+	})
+
+	return nil
+}
+
+func (in *NodePoolSpec) convertTo(ctx context.Context, v1beta1np *v1beta1.NodePoolSpec) {
+	v1beta1np.Weight = in.Weight
+	v1beta1np.Limits = v1beta1.Limits(in.Limits)
+	in.Disruption.convertTo(&v1beta1np.Disruption)
+	in.Template.convertTo(ctx, &v1beta1np.Template)
+}
+
+func (in *Disruption) convertTo(v1beta1np *v1beta1.Disruption) {
```

Comment:
What happens here if the disruption block is `nil`?

Reply:
How can you have a `nil` disruption block? We have default values set, no?